### PR TITLE
consolidate a bunch of FakePodControl testclients into a shared one

### PR DESCRIPTION
**Pull request by mikedanese**

### CODE REVIEW

File: pkg/controller/controller_utils.go [hunk 1]
```diff
@@ -312,6 +312,50 @@ func (r RealPodControl) DeletePod(namespace, podID string) error {
 	return r.KubeClient.Pods(namespace).Delete(podID, nil)
 }
 
+type FakePodControl struct {
+	sync.Mutex
+	Templates     []api.PodTemplateSpec
+	DeletePodName []string
+	Err           error
+}
```
Comment (pmorie) [inline pkg/controller/controller_utils.go]: How about an assertion that this implements `PodControlInterface` here?

**Comment by pmorie** [comment]
@mikedanese LGTM modulo assertion; looks like you caught em' all.

File: pkg/controller/controller_utils.go (hunk 1 above)
Comment (mikedanese) [inline pkg/controller/controller_utils.go]: Done https://github.com/kubernetes/kubernetes/pull/14316/files#diff-4287daff8f8d21b4cd2d0786fbf1c956R41

**Comment by mikedanese** [comment]
@k8s-bot ok to test

File: pkg/controller/controller_utils_test.go [hunk 2]
```diff
@@ -38,6 +38,12 @@ import (
 	"k8s.io/kubernetes/pkg/util/sets"
 )
 
+var (
+	// check structs conform to PodControlInterface
```
Comment (pmorie) [inline pkg/controller/controller_utils_test.go]: the idiom is to have these assertions at the declaration site for the type, partially as a hint to hyman readers about what interfaces a type is supposed to implement. Relocate?

**Comment by mikedanese** [comment]
@pmorie Thanks for review. I've made suggest move. PTAL

**Comment by pmorie** [comment]
@mikedanese LGTM

**Comment by pmorie** [comment]
@mikedanese Ugh, rebase needed.

**Comment by pmorie** [comment]
I don't understand why the merge bot didn't pick this one up.

**Pull request merged**